docs/user-guide/operations/performance-tuning-tips.md

```diff
@@ -0,0 +1,163 @@
+
+GreptimeDB provides various metrics to help monitor and troubleshoot performance issues. The official repository provides [Grafana dashboard templates](https://github.com/GreptimeTeam/greptimedb/tree/main/grafana) for both standalone and cluster modes.
+
+## Query
+
+### Metrics
+
+The following metrics help diagnose query performance issues:
+| Metric | Type | Description |
+|---|---|---|
+| greptime_mito_read_stage_elapsed_bucket | histogram | The elapsed time of different phases of a query in the storage engine. |
+| greptime_mito_cache_bytes | gauge | Size of cached contents |
+| greptime_mito_cache_hit | counter | Total count of cache hit |
+| greptime_mito_cache_miss | counter | Total count of cache miss |
+
+
+### Using cache for object stores
+
+It's highly recommended to enable the object store read cache and the write cache in the storage engine. This could reduce query time by more than 10 times.
+
+The read cache stores objects or ranges on the local disk to avoid fetching the same range from the remote again. The following example shows how to enable the read cache for S3.
+- The `cache_path` is the directory to store cached objects.
+- The `cache_capacity` is the capacity of the cache.
+
+```toml
+[storage]
+type = "S3"
+bucket = "ap-southeast-1-test-bucket"
+root = "your-root"
+access_key_id = "****"
+secret_access_key = "****"
+endpoint = "https://s3.amazonaws.com/"
+region = "your-region"
+cache_path = "/path/to/s3cache"
+cache_capacity = "10G"
+```
+
+The write cache acts as a write-through cache that stores files on the local disk before uploading them to the object store. This reduces the first query latency. The following example shows how to enable the write cache.
+- The `enable_experimental_write_cache` flag enables the write cache
+- The `experimental_write_cache_size` sets the capacity of the cache
+- The `experimental_write_cache_path` sets the path to store cached files. It is under the data home by default.
+- The `experimental_write_cache_ttl` sets the TTL of the cached files.
+
+
+```toml
+[[region_engine]]
+[region_engine.mito]
+enable_experimental_write_cache = true
+experimental_write_cache_size = "10G"
+experimental_write_cache_ttl = "8h"
+# experimental_write_cache_path = "/path/to/write/cache"
+```
+
+### Enlarging cache size
+
+You can monitor the `greptime_mito_cache_bytes` and `greptime_mito_cache_miss` metrics to determine if you need to increase the cache size. The `type` label in these metrics indicates the type of cache.
+
+If the `greptime_mito_cache_miss` metric is consistently high and increasing, or if the `greptime_mito_cache_bytes` metric reaches the cache capacity, you may need to adjust the cache size configurations of the storage engine.
+
+Here's an example:
+
+```toml
+[[region_engine]]
+[region_engine.mito]
+experimental_write_cache_size = "10G"
+sst_meta_cache_size = "128MB"
+vector_cache_size = "512MB"
+page_cache_size = "512MB"
+selector_result_cache_size = "512MB"
```

Comment:
Please explain all these cache settings. What do they mean, and what are the recommended values?

Reply:
done